R/get_coding_ssm_status.R

```diff
@@ -41,49 +80,108 @@
 #' #all lymphoma genes from bundled NHL gene list
 #' coding_tabulated_df = get_coding_ssm_status()
 #'
-get_coding_ssm_status = function(gene_symbols,
-                                 these_samples_metadata,
-                                 from_flatfile = TRUE,
-                                 augmented = TRUE,
-                                 min_read_support = 3,
-                                 maf_path = NULL,
-                                 maf_data,
-                                 include_hotspots = TRUE,
-                                 keep_multihit_hotspot = FALSE,
-                                 recurrence_min = 5,
-                                 this_seq_type = "genome",
-                                 projection = "grch37",
-                                 review_hotspots = TRUE,
-                                 genes_of_interest = c("FOXO1", "MYD88", "CREBBP"),
-                                 genome_build = "hg19",
-                                 include_silent = TRUE){
+get_coding_ssm_status = function(
+    gene_symbols,
+    these_samples_metadata,
+    from_flatfile = TRUE,
+    augmented = TRUE,
+    min_read_support = 3,
+    maf_path = NULL,
+    maf_data,
+    include_hotspots = TRUE,
+    keep_multihit_hotspot = FALSE,
+    recurrence_min = 5,
+    this_seq_type = "genome",
+    projection = "grch37",
+    review_hotspots = TRUE,
+    genes_of_interest = c("FOXO1", "MYD88", "CREBBP"),
+    genome_build = "hg19",
+    include_silent = FALSE,
+    include_silent_genes
+){
+  
+  message(
+    "Using the GAMBLR.results version of this function."
+  )
 
   if(missing(gene_symbols)){
     message("defaulting to all lymphoma genes")
     gene_symbols = pull(GAMBLR.data::lymphoma_genes, Gene)
   }
 
+  if(!missing(include_silent_genes)){
+    message(
+        "Combining genes specified with gene_symbols and include_silent_genes"
+    )
+    gene_symbols <- c(
+        gene_symbols,
+        include_silent_genes
+    ) %>%
+    unique()
+  }
+
   if(missing(these_samples_metadata)){
     these_samples_metadata = get_gambl_metadata()
   }
 
+
+  coding_class <- c(
+    "Frame_Shift_Del", "Frame_Shift_Ins", "In_Frame_Del",
+    "In_Frame_Ins", "Missense_Mutation", "Nonsense_Mutation",
+    "Nonstop_Mutation", "Splice_Region", "Splice_Site",
+    "Targeted_Region", "Translation_Start_Site"
+  )
+
+  if(include_silent){
```

Comment:
Please change to "Including Synonymous variants for all genes" to be a bit more explicit.